EREGCSC-1562 - Implement Snyk

KaushikSamtek opened the pull request:
Resolves # [EREGCSC-1562](https://jiraent.cms.gov/browse/EREGCSC-1562)

**Description** 
This PR is for Snyk implementation. It contains the workflow to run the scan and then calls the service that creates the Jira tickets.

**This pull request changes...**

This pull request changes none of the existing files. Only one new workflow file was added.

**Steps to manually verify this change**

When a pull request is made to main, there will be a Snyk run in the actions tab.

KaushikSamtek commented:
Sorry, I just had to remove push from the title for snyk. No major changes were made.

thwalker6 commented on .github/workflows/snyk-test.yml:
```diff
@@ -0,0 +1,59 @@
+    
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
```
We currently use checkout@v3 on our application since that is the latest version.  Please update.

thwalker6 reviewed (changes requested):
Please update the github actions for checkout to v3.

KaushikSamtek commented:
I have updated the version to v3. I also saw this branch was a commit behind main so I updated that also and not it is ready to go. Let me know if there is anything else

thwalker6 reviewed (approved):
lgtm
